components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emits an event each time a new link is added to a subreddit",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: {
+      propDefinition: [reddit, "subreddit"],
+    },
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for new links on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minutes.
+      },
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emits 10 sample events on the first run during deploy.
+      let before = null;
+      try {
+        var reddit_things = await this.reddit.getNewSubredditLinks(
+          null,
+          this.subreddit,
+          10
+        );
+      } catch (err) {
+        if (
```

Comment:
I'd recommend using Lodash's get method like you are elsewhere to just grab `err.response.status`, since that'll return undefined if any intermediate property is missing.

The 404 behavior also isn't working for me. I entered a random string for the subreddit ("lkjsdflkjdsf") and I saw no error in the logs.